rand_distr/src/gumbel.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2021 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! The Gumbel distribution.
+
+use crate::{Distribution, OpenClosed01};
+use core::fmt;
+use num_traits::Float;
+use rand::Rng;
+
+/// Samples floating-point numbers according to the Gumbel distribution
+///
+/// This distribution has density function:
+/// `f(x) = exp(-(z + exp(-z))) / σ`, where `z = (x - μ) / σ`
```

Comment:
I guess it's evident enough, but it would be nice if there was some consistency between parameter names and those passed to the constructor. And...

- `μ` is often used for mean but in this distribution for mode; okay I guess. WP uses μ; Wolfram uses α.
- `σ` is best known to represent standard deviation, and here is used for scale. Maybe also okay. Most descriptions of the distribution that I can find use `β`.

Reply:
I'm good with other symbols as well; typically I've used μ and σ for location-scale families in general (even if the parameters don't map directly to a mean and standard deviation) because it's less symbols to keep track of. But that's just personal style / habit and would be fine using others, especially if it could lead to more consistency in the code base.

Reply:
I'm fine with either set of symbols, but I agree that they should be defined, for example by adding "where μ is the location and σ is the scale".